postgres/src/client.rs

```diff
@@ -450,3 +453,18 @@ impl Client {
         self.client.is_closed()
     }
 }
+
+impl GenericConnection for Client {
+    fn execute(&mut self, query: &str, params: &[&(dyn ToSql + Sync)]) -> Result<u64, Error> {
+        self.execute(query, params)
+    }
+    fn query(&mut self, query: &str, params: &[&(dyn ToSql + Sync)]) -> Result<Vec<Row>, Error> {
+        self.query(query, params)
+    }
+    fn prepare(&mut self, query: &str) -> Result<Statement, Error> {
```

Comment:
It seems strange that you can prepare a statement but not use it through this interface.

Reply:
Whoops sorry I didn't even realize that query (and execute) had a generic, I have updated them to include the same trait and generic as the normal implementations

Reply:
That does mean that the trait isn't object safe, if that's relevant for what you're trying to do with it.